skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/services/impl/application/PostServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package fr.uca.cdr.skillful_network.services.impl.application;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import fr.uca.cdr.skillful_network.entities.application.JobOffer;
+import fr.uca.cdr.skillful_network.entities.application.Post;
+import fr.uca.cdr.skillful_network.repositories.application.PostRepository;
+import fr.uca.cdr.skillful_network.services.AuthenticationService;
+import fr.uca.cdr.skillful_network.services.application.PostService;
+
+@Service(value="postService")
+public class PostServiceImpl implements PostService{
+	
+    @Autowired
+	private PostRepository repository;
+    
+    @Autowired
+    private AuthenticationService authenticationService;
+	
+	@Override
+	public Post createPost(Post post) {
+		return this.repository.save(post);
+	}
+
+	@Override
+	public List<Post> getAll() {
+		return this.repository.findAll();
+	}
+
+	@Override
+	public List<Post> getPostsforCurrentUser() {
+		return this.getByUserId(authenticationService.getCurrentUser().getId());
+	}
+
+	@Override
+	public void deletePostById(long id) {
+		 this.repository.deleteById(id);
+	}
+
+	@Override
+	public Post update(long id, String postbodyText, Date dateOfPost, Set<String> files) {
+		final Post post = this.repository.findById(id).orElseThrow(() ->
+		new ResponseStatusException(HttpStatus.NOT_FOUND,
```

Comment:
It is better to let the controller handle exception here, because the controller is responsible of the HTTP layer. I would prefer to have this in the controller, and leave all the HTTP stuff to it, while the service is concerned only by the business.